Ingress/controllers/nginx-third-party/nginx.tmpl

```diff
@@ -0,0 +1,352 @@
+        text             text/plain;
+    }
+
+
+    server_name_in_redirect off;
+    port_in_redirect off;
+
+    ssl_protocols {{ $cfg.SSLProtocols }};
+
+    # turn on session caching to drastically improve performance
+    {{ if $cfg.SSLSessionCache }}
+    ssl_session_cache builtin:1000 shared:SSL:{{ $cfg.SSLSessionCacheSize }};
+    ssl_session_timeout {{ $cfg.SSLSessionTimeout }};
+    {{ end }}
+
+    # allow configuring ssl session tickets
+    ssl_session_tickets {{ if $cfg.SSLSessionTickets }}on{{ else }}off{{ end }};
+
+    # slightly reduce the time-to-first-byte
+    ssl_buffer_size {{ $cfg.SSLBufferSize }};
+
+    {{ if not (empty $cfg.SSLCiphers) }}
+    # allow configuring custom ssl ciphers
+    ssl_ciphers '{{ $cfg.SSLCiphers }}';
+    ssl_prefer_server_ciphers on;
+    {{ end }}
+
+    {{ if not (empty .sslDHParam) }}
+    # allow custom DH file http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_dhparam
+    ssl_dhparam {{ .sslDHParam }};
+    {{ end }}
+
+    # Custom error pages using
+    proxy_intercept_errors on;
+    error_page 403 @custom_403;
+    error_page 404 @custom_404;
+    error_page 405 @custom_405;
+    error_page 408 @custom_408;
+    error_page 413 @custom_413;
+    error_page 500 @custom_500;
+    error_page 501 @custom_501;
+    error_page 502 @custom_502;
+    error_page 503 @custom_503;
+    error_page 504 @custom_504;
+
+{{range $ing := .ingresses.Items}}{{range $rule := $ing.Spec.Rules}}
+    server {
+        listen 80;
+        server_name {{ $rule.Host }};
+        vhost_traffic_status_filter_by_host on;        
+{{ range $path := $rule.HTTP.Paths }}
+
+{{ template "LOCATION" (dict "cfg" $cfg "path" $path "ing" $ing "defErrorSvc" $defErrorSvc) }}
+{{ end }}{{/* end loop rule paths */}}
+{{ if $defErrorSvc }}{{ template "CUSTOM_ERRORS" (dict "cfg" $cfg "defErrorSvc" $defErrorSvc) }}{{ end }}
+    }
+
+{{ $sslCertificate := getSSLHost $rule.Host $sslCertificates }}
+{{ if $sslCertificate.Valid }}
+    server {
+        listen 443 ssl http2; 
+        ssl on;
+
+        server_name {{ $rule.Host }};
+        vhost_traffic_status_filter_by_host on;
+
+        ssl_certificate {{ $sslCertificate.Cert }};
+        ssl_certificate_key {{ $sslCertificate.Key }};
+
+{{ if $cfg.EnableHSTS }}
+        add_header Strict-Transport-Security $sts always;
+{{ end }}
+
+{{ range $path := $rule.HTTP.Paths }}
+{{ template "LOCATION" (dict "cfg" $cfg "path" $path "ing" $ing "defErrorSvc" $defErrorSvc) }}
+{{ end }}{{/* end loop rule paths */}}
+{{ if $defErrorSvc }}{{ template "CUSTOM_ERRORS" (dict "cfg" $cfg "defErrorSvc" $defErrorSvc) }}{{ end }}
+    }
+{{ end }}{{/* end if valid certificate */}}
+{{ end }}{{/* end loop Ingress rules */}}{{ end }}{{/* end loop Ingresses */}}
+
+# Idea 1 (done)
+#   iterate []SSLCertificate list
+#   for each check if exists one or more Ingress rules associated with the 
+#   hostname in the rule to create a server listening in port 443.
+#
```

Comment:
without actually parsing this entire config: are you doing both these? if not, which one? I'd modify the comment to describe what you're doing, and state the other as an alternative.